web/src/main/java/ua/com/vertex/logic/CourseLogicImpl.java

```diff
@@ -7,23 +7,56 @@
 import ua.com.vertex.beans.Course;
 import ua.com.vertex.dao.interfaces.CourseDaoInf;
 import ua.com.vertex.logic.interfaces.CourseLogic;
+import ua.com.vertex.utils.DataNavigator;
 
+import java.sql.SQLException;
 import java.util.List;
 
 @Service
 public class CourseLogicImpl implements CourseLogic {
-
-    private final CourseDaoInf courseDaoInf;
     private static final Logger LOGGER = LogManager.getLogger(CourseLogicImpl.class);
+    private final CourseDaoInf courseDao;
+
+    @Override
+    public List<Course> getCoursesPerPages(DataNavigator dataNavigator) {
+        List<Course> courses = null;
+        if (dataNavigator.getCurrentNumberPage() == dataNavigator.getNextPage()) {
+            dataNavigator.setCurrentNumberPage(1);
+            dataNavigator.setNextPage(1);
+        } else {
+            dataNavigator.setCurrentNumberPage(dataNavigator.getNextPage());
+        }
+
+        try {
+            courses = courseDao.getAllCourses(dataNavigator);
+            int dataSize = courseDao.getQuantityCourses();
+
+            dataNavigator.setDataSize(dataSize);
```

Comment:
some strange logic here that also changes incoming parameter. This approach is called "side effect". I do remember time when I was able to kill for that...